ocs_ci/ocs/resources/bucket_notifications_manager.py

```diff
@@ -0,0 +1,362 @@
+        logger.info("Guaranteed bucket logs have been enabled")
+
+    def disable_bucket_logging_on_cr(self):
+        """
+        Unset the bucket notifications feature on the NooBaa CR
+        """
+        logger.info("Disabling bucket notifications on the NooBaa CR")
+
+        try:
+            patch_params = [
+                {
+                    "op": "replace",
+                    "path": NOTIFS_YAML_PATH_NB_CR,
+                    "value": None,
+                },
+            ]
+            self.nb_config_resource.patch(
+                params=json.dumps(patch_params),
+                format_type="json",
+            )
+        except CommandFailed as e:
+            if "not found" in str(e).lower():
+                logger.info("The bucketNotifications field was not found")
+            else:
+                logger.error(f"Failed to disable bucket notifications: {e}")
+                raise e
+        wait_for_pods_to_be_running(
+            pod_names=[constants.NOOBAA_CORE_POD],
+            timeout=60,
+            sleep=10,
+        )
+        logger.info("Bucket notifications have been disabled")
+
+    def create_kafka_topic(self, topic_name=""):
+        """
+        Create a Kafka topic
+
+        Args:
+            name(str|optional): Name of the Kafka topic
+
+        Returns:
+            str: Name of the created Kafka topic
+        """
+        topic_name = topic_name or create_unique_resource_name(
+            resource_description="nb-notif", resource_type="kafka-topic"
+        )
+        topic = self.amq.create_kafka_topic(topic_name)
+        self.kafka_topics.append(topic)
+        return topic_name
+
+    def create_kafka_conn_secret(self, topic):
+        """
+        Create secret from a JSON file that
+        defines the Kafka connection configuration
+
+        Args:
+            topic(str): Name of the Kafka topic
+
+        Returns:
+            secret_ocp_obj: OCP instance of the created secret
+            conn_file_name: Name of the JSON file
+        """
+        conn_name = create_unique_resource_name(
+            resource_description="nb-notif", resource_type="kafka-conn"
+        )
+        secret_name = conn_name + "-secret"
+        conn_file_name = ""
+
+        kafka_conn_config = {
+            "metadata.broker.list": "my-cluster-kafka-bootstrap.myproject.svc.cluster.local:9092",
+            "notification_protocol": "kafka",
+            "topic": topic,
+            "name": conn_name,
+        }
+
+        with tempfile.NamedTemporaryFile(
+            mode="w+", prefix="kafka_conn_", suffix=".json", delete=True
+        ) as conn_file:
+            conn_file_name = os.path.basename(conn_file.name)
+            conn_file.write(json.dumps(kafka_conn_config))
+            conn_file.flush()  # Ensure that the data is written
+
+            OCP().exec_oc_cmd(
+                f"create secret generic {secret_name} --from-file={conn_file.name} -n {self.namespace}"
+            )
+
```

Comment:
Please close the file once secret is created to release handler

Reply:
The use of `with tempfile.NamedTemporaryFile(...) as conn_file` guarantees the closing of the file after this block (or if an exception interrupts it for any reason)

Reply:
Ohh I missed with keyword. Thanks for the clarification